res/schema.xml

```diff
@@ -6,6 +6,16 @@ revisions of the schema on a version upgrade.
 
 DO NOT EDIT THIS FILE OR YOU WILL BREAK YOUR MIXXX LIBRARY AND LOSE YOUR
 METADATA
+
+Backwards compatible migrations are required to be "reappliable" safely
+after the database has been opened and used with an older Mixxx version!!
+If this is not possible then a migration does not qualify to be backwards
+compatible. In this case the min_compatible attribute has to be set to
+the current version, i.e. the same version as the migration.
+
+Errors when adding table columns that already exist when reapplying a
```

Comment:
Did you think about splitting up schema upgrade and data migration?
What happens if it is cbuned in one query? Is sqlite smart enough to handle the case or will both fail with leaving not migrated data behind?

Reply:
Someone else may do that. I won't.

Reply:
How is the current situation?

Reply:
- All (at least the recent ones) migrations include conflict resolution statements (IF NOT EXIST/OR IGNORE) that allow them to be re-applied
- Errors when re-adding columns during a migration, because they could not be prevented in SQL

Critical are only migrations that convert data from one representation into another.

Sorry, but I am not able to come up with a full-featured, all-inclusive solution that considers **all** possible future migrations.